docs/content/en/docs/Design/simd.md

```diff
@@ -0,0 +1,175 @@
+
+For example, the common "rotate-and-reduce" pattern which results in each element containing the sum/product/etc of the original vector can be expressed as:
+```llvm
+%tensor = tensor.from_elements %i1, %i2, %i3, %i4, %i5, %i6, %i7, %i8 : tensor<8xi16>
+%0 = tensor_ext.rotate %tensor, %c4 : tensor<8xi16>, index
+%1 = arith.addi %tensor, %0 : tensor<8xi16>
+%2 = tensor_ext.rotate %1, %c2 : tensor<8xi16>, index
+%3 = arith.addi %1, %2 : tensor<8xi16>
+%4 = tensor_ext.rotate %3, %c1 : tensor<8xi16>, index
+%5 = arith.addi %3, %4 : tensor<8xi16>
+```
+The `%cN` and `%iN`, which are defined as `%cN = arith.constant N : index` and `%iN = arith.constant N : i16`, respectively, have been omitted for readability.
+
+## Intended Usage
+The `-heir-simd-vectorizer` pipeline transforms a program consisting of loops and index-based accesses into tensors (e.g., `tensor.extract` and `tensor.insert`) into one consisting of SIMD operations (including rotations) on entire tensors.
+While its implementation does not depend on any FHE-specific details or even the Secret dialect, this transformation is likely only useful when lowering a high-level program to an arithmetic-circuit-based FHE scheme (e.g., B/FV, BGV, or CKKS).
+The `-mlir-to-openfhe-bgv` pipeline demonstrates the intended flow: augmenting a high-level program with `secret` annotations, then applying the SIMD optimization (and any other high-level optimizations) before lowering to BGV operations and then exiting to OpenFHE.
+
+> **Warning**
+> The current SIMD vectorizer pipeline supports only one-dimensional tensors.
+> As a workaround, one could reshape all multi-dimensional tensors into one-dimensional tensors, but MLIR/HEIR currently do not provide a pass to automate this process.
+
+Since the optimization is based on heuristics, the resulting program might not be optimal or could even be worse than a trivial realization that does not use ciphertext packing.
+However, well-structured programs generally lower to reasonable batched solutions, even if they do not achieve optimal batching layouts.
+For common operations such as matrix-vector or matrix-matrix multiplications, state-of-the-art approaches require advanced packing schemes that might map elements into the ciphertext vector in non-trivial ways (e.g., diagonal-major and/or replicated).
+The current SIMD vectorizer will never change the arrangement of elements inside an input tensor and therefore cannot produce the optimal approaches for these operations.
+
+Note, that the SIMD batching optimization is different from, and significantly more complex than, the Straight Line Vectorizer (`-straight-line-vectorize` pass), which simply groups  [ElementwiseMappable](https://mlir.llvm.org/docs/Traits/#elementwisemappable) operations that agree in operation name and operand/result types into vectorized/tensorized versions.
```

Comment:
One question I have that I am happy to add an answer to when I understand is when you want to use one or the either. (Or is it even possible to use straight-line-vectorize to tensor-ify operations and then heir-simd-vectorize after? probably doesn't do anything useful)